libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/NavigationOptionsProvider.kt

```diff
@@ -0,0 +1,16 @@
+package com.mapbox.navigation.core.lifecycle
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.options.NavigationOptions
+
+/**
+ * Represents a function that returns [NavigationOptions]
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+fun interface NavigationOptionsProvider {
+
+    /**
+     * Generates [NavigationOptions].
+     */
+    fun createNavigationOptions(): NavigationOptions
```

Comment:
By the way we intentionally removed this type of interface.

There was one for android auto https://github.com/mapbox/1tap-android/pull/1085
And then it was part of the original proposal https://github.com/mapbox/mapbox-navigation-android/pull/5112

From discussions, we worked to remove it. So I'm curious why it is required to add it again 🤔